crates/forge-runner/src/forge_runtime_extension.rs

```diff
@@ -54,254 +47,112 @@ use starknet::signers::SigningKey;
+{
+    type Runtime = ContractExecutionSyscallHandler<'a>;
 
-    /// Trait function to store hint in the hint processor by string.
-    fn compile_hint(
-        &self,
-        hint_code: &str,
-        _ap_tracking_data: &ApTracking,
-        _reference_ids: &HashMap<String, usize>,
-        _references: &[HintReference],
-    ) -> Result<Box<dyn Any>, VirtualMachineError> {
-        Ok(Box::new(self.hints[hint_code].clone()))
+    fn get_extended_runtime_mut(&mut self) -> &mut ContractExecutionSyscallHandler<'a> {
+        &mut self.extended_runtime
     }
-}
 
-impl TestExecutionSyscallHandler<'_> {
-    #[allow(clippy::trivially_copy_pass_by_ref, clippy::too_many_arguments)]
-    pub fn execute_cheatcode_hint(
-        &mut self,
-        vm: &mut VirtualMachine,
-        _exec_scopes: &mut ExecutionScopes,
-        selector: &BigIntAsHex,
-        input_start: &ResOperand,
-        input_end: &ResOperand,
-        output_start: &CellRef,
-        output_end: &CellRef,
-        contracts: &HashMap<String, StarknetContractArtifacts>,
-        environment_variables: &HashMap<String, String>,
-    ) -> Result<(), HintError> {
-        // Parse the selector.
-        let selector = &selector.value.to_bytes_be().1;
-        let selector = std::str::from_utf8(selector).map_err(|_| {
-            CustomHint(Box::from(
-                "Failed to parse the  cheatcode selector".to_string(),
-            ))
-        })?;
-
-        // Extract the inputs.
-        let input_start = extract_relocatable(vm, input_start)?;
-        let input_end = extract_relocatable(vm, input_end)?;
-        let inputs = vm_get_range(vm, input_start, input_end)
-            .map_err(|_| CustomHint(Box::from("Failed to read input data".to_string())))?;
-
-        self.match_cheatcode_by_selector(
-            vm,
-            selector,
-            inputs,
-            output_start,
-            output_end,
-            contracts,
-            environment_variables,
-        )
-        .map_err(Into::into)
+    fn get_extended_runtime(&self) -> &ContractExecutionSyscallHandler<'a> {
+        &self.extended_runtime
     }
 
-    #[allow(
-        unused,
-        clippy::too_many_lines,
-        clippy::trivially_copy_pass_by_ref,
-        clippy::too_many_arguments
-    )]
-    fn match_cheatcode_by_selector(
+    #[allow(clippy::too_many_lines)]
+    fn handle_cheatcode(
         &mut self,
-        vm: &mut VirtualMachine,
         selector: &str,
         inputs: Vec<Felt252>,
+        vm: &mut VirtualMachine,
         output_start: &CellRef,
         output_end: &CellRef,
-        contracts: &HashMap<String, StarknetContractArtifacts>,
-        environment_variables: &HashMap<String, String>,
-    ) -> Result<(), EnhancedHintError> {
+    ) -> Result<CheatcodeHadlingResult, EnhancedHintError> {
         let mut buffer = MemBuffer::new_segment(vm);
         let result_start = buffer.ptr;
```

Comment:
In #1209 will change it to not use any buffer and return array of felts instead.